### PR TITLE
Adjust support page and remove billing for self-hosted

### PR DESCRIPTION
**Pull request by adamconnelly**
# Description of the change

I've made some initial adjustments to the support page for self-hosted. It will probably still need more work, but I've tried to at least remove misleading things like the mention of different plans from self-hosted.

I've also removed the billing pages for now because they are heavily geared towards SaaS.

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

**Review by peterdeme — APPROVED** [review]
lgtm

**Review by michalg9 — APPROVED** [review]

File: docs/product/support/README.md [hunk 1]
```diff
@@ -4,30 +4,40 @@ description: Information about Spacelift support options.
...
 ## Contacting Support
 
+{% if is_saas() %}
 | Plan           | How to get support                                           |
 | -------------- | ------------------------------------------------------------ |
 | **Free**       | Open a conversation in the Spacelift chat widget             |
 | **Cloud**      | Open a support ticket in the shared Slack channel (preferred, when available) or open a conversation in the chat widget in the bottom-right corner of the screen |
 | **Enterprise** | Open a support ticket in the shared Slack channel (preferred, when available) or open a conversation in the chat widget in the bottom-right corner of the screen |
+{% else %}
+For technical questions related to the Spacelift product, open a support ticket in the shared Slack channel (preferred, when available).
```
Comment (pawelpiwosz) [inline docs/product/support/README.md:30]: do we have any other way of contact? If yes, maybe we should mention it.

Comment (adamconnelly) [reply docs/product/support/README.md:30]: Good point. We'll discuss this internally and tweak it in a separate change.

Comment (adamconnelly) [reply docs/product/support/README.md:30]: The trouble is that the chat widget isn't available in self-hosted right now.

**Pull request merged**